File: 1710. Maximum Units on a Truck.py
##ss
class Solution:
    def maximumUnits(self, boxTypes: List[List[int]], truckSize: int) -> int:
        
        boxTypes = sorted(boxTypes, reverse = True, key = lambda x:(x[1]))
        ans = 0
        for x in range(len(boxTypes)):
            #print(ans,truckSize)
            if boxTypes[x][0] < truckSize:
                ans+= boxTypes[x][0] * boxTypes[x][1]
                truckSize-= boxTypes[x][0]
                
            else:
                ans += truckSize * boxTypes[x][1]
                break
                
        return ans
        

File: cpp_solutions/1710. Maximum Units on a Truck.cpp
class Solution {
public:
    int maximumUnits(vector<vector<int>>& boxTypes, int truckSize) {
        
        sort(boxTypes.begin(),boxTypes.end(),comp);
        //cout << boxTypes[0][1];
        int ans = 0;
        int i = 0;
        while (i<boxTypes.size() && truckSize > 0) {
            if (boxTypes[i][0] <= truckSize) {
                truckSize-=boxTypes[i][0];
                ans+=boxTypes[i][0] * boxTypes[i][1];
            }
            else {
                ans+=boxTypes[i][1] * truckSize;
                truckSize = 0;
            }
            i+=1;
            //cout << ans << " ";
        }
        return ans;
    }
    
    static bool comp(const vector<int>& a, const vector<int>& b) {
        return a[1] > b[1];
    }
};
